DEVELOPING.adoc

```diff
@@ -79,11 +79,15 @@ If new Kiali Server resources are added or removed, you must make appropriate ch
 
 Below are checklists that developers should consult when making various kinds of changes.
 
+#### Maintain Backward Compatibility
```

Comment:
@jmazzitelli just to make sure I understand correctly:
"Backward compatibility" here means having new versions of the **Operator** still working with all supported versions of **Kiali**, right?

You take the example of permissions and roles but I guess the same is true with, say, config changes? If we take the example of what we worked on previously, amending the `custom_dashboard` config: does it mean we should actually have kept the old `custom_prometheus_url` config alive in the CR field (marking it deprecated)? Should we reintroduce it in the documentation?

Reply:
@jotak I used the roles/permissions as an example because that was the one that made me realize this problem. It affected the permissions given to the operator (the operator needed to create a role for the v1.12 Kiali Server, but the operator was no longer given permissions to give out that role because we removed those permissions).

But, in general, yes, this might be something we need to watch out for when changing config. Certain things may not be a problem, but we just have to remember to think about it. And there are certain things we could work around by updating the versioned Ansible roles (i.e. backporting changes to the `v1.0`, `v1.12` and/or `v1.24` Ansible roles as well as changing the `default` one) so long as that version of Kiali Server is also updated to match (so we would need to make changes to the kiali/kiali repo's version branch to fix the server, too). If we backported that `custom_dashboard` work to the v1.12 branch of the kiali/kiali repo, (to fix the Server), then all we would have to do is change the v1.12 Ansible roles to match. Did you backport that custom_dashboard stuff? If not, we should do that OR we should document that custom_prometheus_url is needed in your Kiali CR for older versions.

In that specific case of `custom_prometheus_url`, I think we are OK because the operator itself didn't use or even look at `custom_prometheus_url` in the older playbooks. So the operator didn't need it. And since the operator passes down any config from the Kiali CR to the ConfigMap, you can still put `custom_prometheus_url` directly in your Kiali CR and the Kiali Server will get it in its ConfigMap. So older Kiali Servers that look for it will see it.